src/main/java/core/basesyntax/service/ParseData.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ParseData {
+    private static final String DATE_REGEXP = "\\d{4}-\\d{2}-\\d{2}";
+    private static final String HEADER = "type,fruit,quantity,date";
+
+    public List<Record> parseRecords(List<String> stringList) {
+        if (removeHeader(stringList)) {
+            List<Record> recordList = new ArrayList<>();
+            for (String string : stringList) {
+                string = string.trim();
+                if (string.matches("[sbr],[a-z]{3,},\\d+," + DATE_REGEXP)) {
+                    recordList.add(parseSingleRecord(string));
+                } else if (!string.equals("type,fruit,quantity,date")) {
+                    throw new RuntimeException("Incorrect line: " + string);
+                }
+            }
+            return recordList;
+        }
+        throw new RuntimeException("Header line [" + HEADER + "] has not been found");
+    }
+
+    private boolean removeHeader(List<String> stringList) {
```

Comment:
I think  - this a bit complecated - I would just ignore first line. but certainly someone could pass file with no header or different header, so you might be right to check it.